File: python/step/start.py
from typing import Tuple, Optional

from lead import Lead


def start(state: int) -> Tuple[int, Optional[Lead]]:
    user_input = input("Enter a starting number (or press Enter for default 0): ")
    if not user_input:
        print(f"Starting count from default: {state}")
        return state, Lead.COUNT
    try:
        initial_count = int(user_input)
        state = initial_count
        print(f"Starting count from: {state}")
        return state, Lead.COUNT
    except ValueError:
        print("Invalid input. Please, try again.")
        return state, Lead.START

File: ruby/step/start.rb
def start(state)
  loop do
    print "Enter a starting number (or press Enter for default 0): "
    user_input = gets.chomp

    if user_input.empty?
      puts "Starting count from default: #{state}"
      return [state, :count]
    end

    initial_count = Integer(user_input)
    state = initial_count
    puts "Starting count from: #{state}"
    return [state, :count]
  rescue ArgumentError
    puts "Invalid input. Please enter a valid number or press Enter."
  end
end
